src/analyses/apron/relationAnalysis.apron.ml

```diff
@@ -158,13 +158,14 @@ struct
         {st' with rel = rel''}
       )
     | (Mem v, NoOffset) ->
-      begin match ask.f (Queries.MayPointTo v) with
-        | ad when Queries.AD.is_top ad -> st
-        | ad ->
-          let mvals = Queries.AD.to_mval ad in
-          let ass' = List.map (fun mval -> assign_to_global_wrapper ask getg sideg st (ValueDomain.Addr.Mval.to_cil mval) f) mvals in
-          List.fold_right D.join ass' (D.bot ())
-      end
+      let ad = ask.f (Queries.MayPointTo v) in
+      Queries.AD.fold (fun addr acc ->
+          match addr with
+          | ValueDomain.Addr.Addr mval ->
+            D.join acc (assign_to_global_wrapper ask getg sideg st (ValueDomain.Addr.Mval.to_cil mval) f)
+          | UnknownPtr | NullPtr | StrPtr _ ->
+            D.join acc st (* Ignore assign *)
```

Comment:
Is simply ignoring the assignment in this case consistent with what base does?

Reply:
As far as I can tell, yes (in `set`):
https://github.com/goblint/analyzer/blob/71e21b27973c78ef58c2567e314e128cfbb10564/src/analyses/base.ml#L1579-L1582
Just a bit more explicit here that we're ignoring unknown and NULL pointers.